nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -764,7 +796,15 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
                                             }
                                             currentValue = dest;
                                         } else if (currentValue instanceof String) {
-                                            currentValue = ((String) currentValue).getBytes(StandardCharsets.UTF_8);
+                                            final String stringValue = (String) currentValue;
+
+                                            if (BINARY_STRING_FORMAT_BASE64.getValue().equals(binaryStringFormat)) {
+                                                currentValue = BaseEncoding.base64().decode(stringValue);
```

Comment:
The standard Java [Base64](https://docs.oracle.com/en/java/javase/21/docs//api/java.base/java/util/Base64.html) class should be used instead of the the Google class.